docs/gen-ai/llm-spans.md

```diff
@@ -65,26 +82,380 @@ These attributes track input data and metadata for a request to an LLM. Each att
+| `tool_call_id` | string | Tool call that this message is responding to. | `call_mszuSIzqtI65i1wAUOE8w5H4` | `Required` |  |
+
+### Choice message
 
-<!-- semconv gen_ai.content.prompt -->
-The event name MUST be `gen_ai.content.prompt`
+This event describes the model-generated chat response message (choice).
+If GenAI model returns multiple choices, each of the message SHOULD be recorded as an individual event.
+
+<!-- semconv gen_ai.choice(full) -->
+The event name MUST be `gen_ai.choice`
 
 | Attribute  | Type | Description  | Examples  | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) | Stability |
 |---|---|---|---|---|---|
-| [`gen_ai.prompt`](/docs/attributes-registry/gen-ai.md) | string | The full prompt sent to an LLM. [1] | `[{'role': 'user', 'content': 'What is the capital of France?'}]` | `Conditionally Required` if and only if corresponding event is enabled | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`gen_ai.system`](/docs/attributes-registry/gen-ai.md) | string | The name of the LLM foundation model vendor. | `openai` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 
-**[1]:** It's RECOMMENDED to format prompts as JSON string matching [OpenAI messages format](https://platform.openai.com/docs/guides/text-generation)
+`gen_ai.system` has the following list of well-known values. If one of them applies, then the respective value MUST be used; otherwise, a custom value MAY be used.
+
+| Value  | Description | Stability |
+|---|---|---|
+| `openai` | OpenAI | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 <!-- endsemconv -->
 
-<!-- semconv gen_ai.content.completion -->
-The event name MUST be `gen_ai.content.completion`
+When response is streamed, instrumentations that report response events MUST reconstruct and report the full message and MUST NOT report individual chunks as events.
+If the request to GenAI model fails with an error before content is received, instrumentation SHOULD report an event with truncated content it was able to receive and if `finish_reason` is not
+yet received, it MUST set it to `error`.
+
+Response event payload has the following fields:
+
+| Body Field | Type | Description | Examples | Requirement Level | Sensitive |
+|---|---|---|---|---|---|
+| `finish_reason` | string | The reason the model stopped generating tokens. | `stop`, `tool_calls`, `content_filter` | `Required` |  |
+| `index` | int | The index of the choice in the list of choices. | `1` | `Required` |  |
+| `message` | [Message](#message-object) | GenAI response message | `{"content":"The OpenAI semantic conventions are available at opentelemetry.io"}` | `Recommended` | ![Mixed](https://img.shields.io/badge/-mixed-orange) |
+
+#### `Message` object
+
+The message structure matches one of the messages defined in this document depending on the role:
```

Comment:
Hi! Shouldn't it only be an assistant message? AFAIK LLMs do not output tool (result) messages, only tool calls, but those are located inside the assistant message.

Reply:
good catch, thanks!